crates/tabby/src/services/chat.rs

```diff
@@ -33,9 +34,45 @@ pub struct Message {
 
 #[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
 pub struct ChatCompletionChunk {
+    id: Uuid,
+    created: u64,
+    object: &'static str,
+    model: &'static str,
+    choices: [ChatCompletionChoice; 1],
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+pub struct ChatCompletionChoice {
+    index: usize,
+    message: ChatCompletionMessageChoice,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+pub struct ChatCompletionMessageChoice {
+    role: String,
+    finish_reason: Option<String>,
     content: String,
 }
 
+impl ChatCompletionChunk {
+    fn new(content: String, id: Uuid, created: u64, last_chunk: bool) -> Self {
+        ChatCompletionChunk {
+            id,
+            created,
+            object: "chat.completion.chunk",
+            model: "TabbyML",
+            choices: [ChatCompletionChoice {
+                index: 0,
+                message: ChatCompletionMessageChoice {
```

Comment:
In chunk object, content should be put inside of `delta` sub field. Example:

https://platform.openai.com/docs/api-reference/chat/streaming